driver/vhci_ude/vhci_urbr.c

```diff
@@ -206,15 +221,31 @@ submit_urbr(purb_req_t urbr)
 
 	urbr->seq_num = ++(vusb->seq_num);
 
-	WdfWaitLockRelease(vusb->lock);
+	WdfSpinLockRelease(vusb->spin_lock);
 
 	status = store_urbr(req_read, urbr);
+	TRD(URBR, "After 'store_urbu()");
 
-	WdfWaitLockAcquire(vusb->lock, NULL);
+	WdfSpinLockAcquire(vusb->spin_lock);
 
 	if (status == STATUS_SUCCESS) {
 		if (urbr->type == URBR_TYPE_URB) {
-			WdfRequestMarkCancelable(urbr->req, urbr_cancelled);
```

Comment:
I'll be glad to try this, because i do not know internals enough (i just tried to follow an example i found):)

Reply:
Here is the link to example i used (see example 2: https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdfrequest/nf-wdfrequest-wdfrequestmarkcancelableex).

Otherwise, i am trying out your change and  i get quite a solid freeze of the system after connecting my USB audio interface and after i change its output volume in the system tray. It is funny, i am already watching this part, because also under my implementation, the system eventually starts crawling but not so hard. The other interesting thing is that i get console error: unknown command in pdu header: 656 - usually that number but not always. From what i could conclude from traces, it has to do something with 'isochronous' requests handling.

Reply:
Also a quick test of USB storage device seems to operate smoothly from driver installation, device connection, its usage (i.e. a file creation), up to its disconnection and driver unload.

Reply:
> unknown command in pdu header: 656

This means something wrong happened. Maybe, a failed ISO transfer broke the protocol buffer. Does your USB audio have the same problem without my suggested modification?

Reply:
Without your suggestion something similar happens but not so quickly:). On the other hand both variants do not differ for the non-error path, so they should behave the same until errors occur. 
Another thing about error case for this code that puzzles me is, should a kind of 'complete_urbr()' be called and also 'submit_urbr_unlink()' at least for the second part?

Momentary my plan is to add more debug output and try to pinpoint the cause of protocol break.